### PR TITLE
Reimplement savestate loading via drag and drop

### PR DESCRIPTION
**Pull request by NarryG**
This PR re-implements the missing DolphinWX functionality where you could drag and drop a savestate onto the render window and it'd load it for you.

All savestate validation is already performed within `State::LoadAs` so I just lifted the drag and drop code from MainWindow and implemented it into RenderWidget.

### CODE REVIEW

File: Source/Core/DolphinQt/RenderWidget.h [hunk 1]
```diff
@@ -36,6 +37,8 @@ class RenderWidget final : public QWidget
   void OnKeepOnTopChanged(bool top);
   void SetFillBackground(bool fill);
   void OnFreeLookMouseMove(QMouseEvent* event);
+  void dragEnterEvent(QDragEnterEvent* event) override;
```
Comment (Techjar) [inline Source/Core/DolphinQt/RenderWidget.h]: Any reason this is in the header? It's not implemented.

**Comment by zackhow** [comment]
@dolphin-emu-bot rebuild

**Comment by NarryG** [comment]
I see lint failed as VS stuck the includes at the bottom and I missed fixing them. Should be fixed.

File: Source/Core/DolphinQt/RenderWidget.h [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 class QMouseEvent;
 class QTimer;
+class DragEnterEvent;
```
Comment (lioncash) [inline Source/Core/DolphinQt/RenderWidget.h]: This is missing a Q prefix

File: Source/Core/DolphinQt/RenderWidget.cpp [hunk 3]
```diff
@@ -80,6 +87,35 @@ void RenderWidget::SetFillBackground(bool fill)
...
+
+  auto path = file_info.filePath();
+
+  if (!file_info.exists() || !file_info.isReadable())
+  {
+    QMessageBox::critical(this, tr("Error"), tr("Failed to open '%1'").arg(path));
+    return;
+  }
+
+  if (file_info.isFile())
```
Comment (lioncash) [inline Source/Core/DolphinQt/RenderWidget.cpp]: This can be inverted so it's not structuring the valid codepath like the invalid ones that precede it.

Comment (NarryG) [reply Source/Core/DolphinQt/RenderWidget.cpp]: This code was lifted from MainWindow.cpp. I can invert it here. Want me to do a pull inverting it in master after?

File: Source/Core/DolphinQt/RenderWidget.h (hunk 2 above)
Comment (BhaaLseN) [reply Source/Core/DolphinQt/RenderWidget.h]: Kinda surprised this compiles if its the wrong prefix. Is it even needed in here?

Comment (NarryG) [reply Source/Core/DolphinQt/RenderWidget.h]: I lifted this code from MainWindow.h. It was included there with no Q prefix.

Comment (NarryG) [reply Source/Core/DolphinQt/RenderWidget.h]: The inclusion of QWidget is all that's required for QDragEnterEvent and QDropEvent to resolve.  I only included it as it was included in MainWindow.h which is where I lifted the code from. It compiled as it's not even required.

**Comment by leoetlino** [comment]
Code LGTM, but could you give a shorter name to your commit title? I don't care about titles being slightly too long, but 140+ is way, way too much.

**Comment by NarryG** [comment]
Done @leoetlino

**Comment by NarryG** [comment]
Actually done (misread initially and thought you said PR not commit message) @leoetlino

**Comment by NarryG** [comment]
Closing this because I originally submitted the PR from master and just goofed up the branch when syncing my fork up, forgetting I had submitted this from master. Will re-submit on a new branch.

**Pull request closed without merging**